### PR TITLE
Make possible to adjust heap sizes via properties

### PR DESCRIPTION
**Pull request by Holmistr**
Aligning with enterprise: https://github.com/hazelcast/hazelcast-enterprise/commit/43554c3561c903bda3baf0b0152d93e2e7cc2ce5

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -705,140 +717,6 @@
             </build>
         </profile>
 
-        <profile>
-            <id>test-coverage-nightly</id>
```
Comment (Donnerbart) [inline pom.xml]: I just confirmed that we're using the `test-coverage` profile for the nightly Sonar OS build.

Comment (pveentjer) [reply pom.xml]: Perhaps we should rename the profile to 'Sonar OS build' to indicate what is being used for?

Comment (Holmistr) [reply pom.xml]: @pveentjer I removed the profile as it's not used at all, it's a left over from the past.

**Comment by Holmistr** [comment]
I have unified the heap settings as much as possible (addressed @Donnerbart 's comment).

**Comment by Holmistr** [comment]
@kwart approved and @Donnerbart 's comments were addressed. I need this PR to be merged ASAP, I'm happy to issue another to address other comments.

**Pull request merged**